include/sound/sof/stream.h

```diff
@@ -85,8 +85,9 @@ struct sof_ipc_stream_params {
 
 	uint32_t host_period_bytes;
 	uint16_t no_stream_position; /**< 1 means don't send stream position */
+	uint16_t cont_update_posn; /**< 1 means continue update stream position */
 
```

Comment:
Could we just take only 1 byte from reserved area. We are wasting so much space to keep a boolean information.

Reply:
@dbaluta yes, I could modify as one byte. Thanks

Reply:
We will set the `cont_update_posn` to one in kernel unconditionally or what is the condition to set it?
Iow, will the use of this new flag going to be:
```
if ()
+ 	if (v->abi_version >= SOF_ABI_VER(3, 21, 0))
+  		msg->cont_update_posn = 1;
```
If it is then I argue that there is no point of having this flag in the first place and just do it unconditionally in firmware?

Reply:
@ujfalusi Currently I propose to set it in the stream callback function pcm_hw_params e.g.
```c
#define CONTINUOUS_UPDATE_POSITION 1

static int mt8195_dsp_pcm_hw_params(struct snd_sof_dev *sdev,
			  struct snd_pcm_substream *substream,
			  struct snd_pcm_hw_params *params,
			  struct sof_ipc_stream_params *ipc_params)
{
	struct sof_ipc_fw_version *v = &sdev->fw_ready.version;

	if (v->abi_version >= SOF_ABI_VER(3, 21, 0))
		ipc_params->cont_update_posn = CONTINUOUS_UPDATE_POSITION;

	return 0;
}
```

Reply:
I think this should be the default across all platforms, I don't really see a reason why this capability needs to be added 4 times when we all have to support the Chrome conformance suite. @cujomalainey any comments?

Reply:
agreed, given we know the current default does not pass the test, lets push this into the core and replace the existing default

Reply:
@yaochunhung can you please add a user for this new capability?